pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -880,12 +881,22 @@ void splitAndOwnBundleOnceAndRetry(NamespaceBundle bundle,
                                     for (NamespaceBundle sBundle : splittedBundles.getRight()) {
                                         Objects.requireNonNull(ownershipCache.tryAcquiringOwnership(sBundle));
                                     }
-                                    updateNamespaceBundles(nsname, splittedBundles.getLeft())
-                                            .thenRun(() -> {
-                                                bundleFactory.invalidateBundleCache(bundle.getNamespaceObject());
-                                                updateFuture.complete(splittedBundles.getRight());
-                                            }).exceptionally(ex1 -> {
-                                        String msg = format("failed to update namespace policies [%s], "
+                                    updateNamespaceBundles(nsname, splittedBundles.getLeft()).thenRun(() -> {
+                                        updateNamespaceBundlesForPolicies(nsname, splittedBundles.getLeft())
+                                                .thenRun(() -> {
+                                                    bundleFactory.invalidateBundleCache(bundle.getNamespaceObject());
+                                                    updateFuture.complete(splittedBundles.getRight());
+                                                }).exceptionally(e -> {
+                                                    String msg = format("failed to update namespace policies [%s], "
+                                                                    + "NamespaceBundle: %s due to %s",
+                                                            nsname.toString(), bundle.getBundleRange(), e.getMessage());
+                                                    LOG.warn(msg);
+                                                    updateFuture.completeExceptionally(
+                                                            new ServiceUnitNotReadyException(msg, e.getCause()));
+                                                    return null;
+                                                });
```

Comment:
Is it possible to remove line889 to line897？It seems same with line898 to line905

Reply:
Fixed,PTAL,thanks! @AnonHxy